syntax = "proto3";

package intellidocs_proto.v2;

message ProcessPDFsRequest {
  repeated string pdf_paths = 1;
}

message ProcessPDFsResponse {
  map<string, string> pdf_keys = 1;
}

message QueryDocumentRequest {
  string pdf_key = 1;
  string user_query = 2;
  int32 top_n = 3;
}

message QueryDocumentResponse {
  message QueryResult {
    string chunk = 1;
    double score = 2;
  }
  repeated QueryResult results = 1;
}

service IntellidocsService {
  rpc ProcessPDFs (ProcessPDFsRequest) returns (ProcessPDFsResponse);
  rpc QueryDocument (QueryDocumentRequest) returns (QueryDocumentResponse);
}
